### PR TITLE
upgrade dependencies in ouroboros-network

### PR DESCRIPTION
**Pull request by EncodePanda**
- [x] carano-predule
- [x] cardano-base
- [x] cardano-ledger-specs

Fixes https://github.com/input-output-hk/ouroboros-network/issues/3025

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs [hunk 1]
```diff
@@ -49,6 +49,7 @@ import           Ouroboros.Consensus.Util.Condense
 import qualified Shelley.Spec.Ledger.API as SL
 import           Shelley.Spec.Ledger.BlockChain as SL (TxSeq (..))
 import qualified Shelley.Spec.Ledger.UTxO as SL (txid)
+import qualified Cardano.Ledger.Core as Core
```
Comment (Jasagredo) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs:52]: Isn't this import redundant?

File: ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs [hunk 2]
```diff
@@ -80,7 +80,7 @@ import qualified Shelley.Spec.Ledger.API as SL
 import qualified Shelley.Spec.Ledger.BaseTypes as SL (Seed (..),
                      mkNonceFromNumber, textToUrl)
 import qualified Shelley.Spec.Ledger.BlockChain as SL (TxSeq (..))
-import qualified Shelley.Spec.Ledger.Coin as SL (DeltaCoin (..))
+import qualified Cardano.Ledger.Coin as SL (DeltaCoin (..))
```
Comment (Jasagredo) [inline ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs:83]: Maybe worth changing the qualified name to something that does not resemble Shelley as this seems to have been moved away from `Shelley.Spec`. And possibly moving the import 8 lines upward to have it sorted.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs [hunk 3]
```diff
@@ -38,6 +38,8 @@ import qualified Cardano.Ledger.Shelley.Constraints as SL
 import           Ouroboros.Consensus.Shelley.Protocol.Crypto (StandardCrypto)
 import qualified Shelley.Spec.Ledger.API as SL
 import qualified Shelley.Spec.Ledger.BaseTypes as SL
+import qualified Shelley.Spec.Ledger.Tx as SL
+import qualified Cardano.Ledger.Core as Core
```
Comment (Jasagredo) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:42]: `Cardano.Ledger.Core` seems to already be in scope with qualified name `LC`.

**Review by Jasagredo — CHANGES_REQUESTED** [review]
Apart from those import remarks, the changes look good to me.

File: ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs (hunk 2 above)
Comment (EncodePanda) [reply ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs:83]: I've forgot to format it :) Crap.
Regarding rename.. Hmm .. I think `SL` stands for "Spec Ledger" but I might be wrong

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs (hunk 3 above)
Comment (EncodePanda) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs]: Yeah, nice catch. I've removed `Tx`

File: ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs (hunk 2 above)
Comment (EncodePanda) [reply ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs]: This got formatted

**Review by Jasagredo — APPROVED** [review]
LGTM :+1:

**Review by nfrisby — APPROVED** [review]
Nice diff!

I'd like a comment on the `Tx` instance if you have the patience for another iteration, but it's not crucial -- we'll revisit that soon enough.

Also, FYI, I renamed the name of the PR.

Approved. Thanks!

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs [hunk 4]
```diff
@@ -66,6 +67,8 @@ type EraCrypto era = Crypto era
   Era polymorphism
 -------------------------------------------------------------------------------}
 
+type instance Core.Tx era = SL.Tx era
```
[the remark this answers is not in the document]
Comment (EncodePanda) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:73]: I've added a comment here and for `Witnesses` equivalence as well

[the remark this answers is not in the document]
Comment (DavidEichmann) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:73]: I have a question: is the Tx type family only ever indexed on a sheley based eras? More specifically, is the Byron era never used in this type level machinery?

[the remark this answers is not in the document]
Comment (DavidEichmann) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:73]: Ah ok the package description says so:
>   Cardano ledgers from the Shelley release onwards share a core basis rooted in
  the Shelley ledger specification. This package abstracts a number of components
  which we expect to be shared amongst all future ledgers implemented around this base.

**Review by DavidEichmann — APPROVED** [review]

**Comment by EncodePanda** [comment]
bors merge

**Pull request merged**